docker/service/src/main/java/uk/nhs/adaptors/scr/fhirmappings/CompositionMapper.java

```diff
@@ -0,0 +1,87 @@
+package uk.nhs.adaptors.scr.fhirmappings;
+
+import static uk.nhs.adaptors.scr.utils.DateUtil.formatDate;
+import static uk.nhs.adaptors.scr.utils.ThrowFhirMappingError.throwFhirMappingErrorMissingFromPayload;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hl7.fhir.r4.model.Composition;
+
+import uk.nhs.adaptors.scr.exceptions.FhirMappingException;
+import uk.nhs.adaptors.scr.models.GpSummary;
+import uk.nhs.adaptors.scr.models.Presentation;
+
+public class CompositionMapper {
+    public GpSummary mapComposition(GpSummary gpSummary, Composition composition) throws FhirMappingException {
+        if (composition == null) {
```

Comment:
I see you check that in many places. What about doing a single check where all those functions are invoked? Also I don't see here you checking for non null `gpSummary`